### DISCUSSION TITLE
device class icons not showing

### DISCUSSION THREAD

**Discussion by arthurEspooner** [cgiesche/streamdeck-homeassistant#211] [General]
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/79245640/22fb7234-f416-49ba-b34c-91ffbce67d23)

the top right office temperature and humidity below it use temp/humidity sensors and seem to provide their own colored icons. most other things just do a default white icon.

i tried adding a template binary sensor with a device class of cold and it results in the icon shown in the bottom left button. it's just the generic circle. it does toggle with the state, but in HA this device class has a specific icon set as seen [here](https://www.home-assistant.io/integrations/binary_sensor/)

is there anyway to get this "device class" icon to pass through?

**Comment by cgiesche** [comment]
Hi,

i found out, what causes your problem. It's a bit complicated:

For many (but unfortunately not all) entity types, home-assistant sends an "icon" property in the event attributes. If this attribut is present, i take the value and display the matching icon (same as in home assistant gui) on the button. Those icons are static and do not change when the state of the entity changes.

BUT: There are some device-classes that have custom icons defined inside the plugin (which overule the way described above). I am doing this to be able to visually show the state of a light, a switch, the weather and so on. binary_sensor is one of them.

If i disable my overrides completely, some entities (lights) do not have any icons at all and most important, you can not determine if a switch or light is on or off).
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/5123180/9c7cb6b1-444a-4213-86e8-aac4faded1e9)

I think i will add a "force use home assistant icon"-option to the button configuration...

**Comment by arthurEspooner** [comment]
it did seem like you were overriding something so i didn't play with it much. if you could add a toggle that would be great.

**Comment by DennisGaida** [comment]
> I think i will add a "force use home assistant icon"-option to the button configuration...

I was just about to ask the same question. I have set a custom icon in HA:
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/2392217/14e4e5c4-5fb3-4770-9bed-5a52fe3ca8ef)

but via the plugin I only see the standard on/off toggle. Is that to be expected and exactly the behavior you talk about here?

![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/2392217/29d11ea8-555a-4b31-9cc8-8229c7236d74)

**Comment by Calzor-Suzay** [comment]
Are the icons used in Stream Deck exposed anywhere?, I have an entity in the 'Climate' domain which shows my lounge temperature but the button icon is blank, I just merely want to copy paste the green one used for other temp gauges to this button manually so they look the same.

BTW I did get the MDI, icon, re coloured it but it then gave it a white background, I fixed that then realised it comes in at totally wrong size and gave up in the hope there's a much easier way to do this.

**Comment by N0thingSpecial** [comment]
Is this behavior still the same? I want to overwrite an icon with "device_class: power" but still get the lightning bolt. Is there some kind of overwrite-feature yet?

**Comment by cgiesche** [comment]
Coming soon: https://github.com/cgiesche/streamdeck-homeassistant/issues/254
